### ISSUE TITLE
Trans extraction breaks when children and defaults nodes are used together

### ISSUE THREAD

**Issue by beheh** [i18next/i18next-scanner#85]
As of 2.6.1, the following JSX:
```jsx
<React.Fragment>
	<Trans defaults="Some <0>{variable}</0>" />
	<Trans>Hello, World!</Trans>
</React.Fragment>
```

results in the following error:

```
{ SyntaxError: Unexpected token (1:19)
    at Parser.pp$4.raise (/opt/application/source/node_modules/acorn-jsx/node_modules/acorn/dist/acorn.js:2756:13)
    at Parser.pp.unexpected (/opt/application/source/node_modules/acorn-jsx/node_modules/acorn/dist/acorn.js:647:8)
    at Parser.pp.jsx_parseIdentifier (/opt/application/source/node_modules/acorn-jsx/inject.js:194:12)
    at Parser.pp.jsx_parseNamespacedName (/opt/application/source/node_modules/acorn-jsx/inject.js:203:21)
    at Parser.pp.jsx_parseElementName (/opt/application/source/node_modules/acorn-jsx/inject.js:218:21)
    at Parser.pp.jsx_parseClosingElementAt (/opt/application/source/node_modules/acorn-jsx/inject.js:305:25)
    at Parser.pp.jsx_parseElementAt (/opt/application/source/node_modules/acorn-jsx/inject.js:327:35)
    at Parser.pp.jsx_parseElement (/opt/application/source/node_modules/acorn-jsx/inject.js:375:17)
    at Parser.parseExprAtom (/opt/application/source/node_modules/acorn-jsx/inject.js:397:23)
    at Parser.pp$3.parseExprSubscripts (/opt/application/source/node_modules/acorn-jsx/node_modules/acorn/dist/acorn.js:2047:19) pos: 19, loc: Position { line: 1, column: 19 }, raisedAt: 20 }
```

and neither are extracted.

Note this only seems to happen when the defaults mode contains a tag using the `<0></0>` syntax.

#### Version
- i18next: <i>11.3.3</i>
- i18next-scanner: <i>2.6.1</i>

#### Configuration
```js
const fs = require("fs");

module.exports = {
	options: {
		// use strings as keys
		nsSeparator: false,
		keySeparator: false,
		// settings
		defaultNs: "frontend",
		lngs: ["en"],
		resource: {
			loadPath: "{{lng}}/{{ns}}.json",
			savePath: "{{lng}}/{{ns}}.json",
		},
		func: false,
		trans: false,
		defaultValue: (language, namespace, key) => key,
	},
	transform(file, enc, done) {
		const parser = this.parser;
		const content = fs.readFileSync(file.path, enc);

		parser.parseFuncFromString(
			content,
			{
				list: ["t"],
			},
			(key, options) => {
				if (["some", "blacklisted", "keys"].indexOf(key) !== -1) {
					return;
				}
				parser.set(key, options);
			},
		);

		parser.parseTransFromString(
			content,
			{
				component: ["Trans"],
				i18nKey: false,
			},
			(key, options) => {
				parser.set(options.defaultValue, options);
			},
		);

		done();
	},
};

```

**Comment by beheh** [comment]
Looks like this is due to how the parsing works, where we first try to find suitable `<Trans>` components via Regexp, and then hand the code off to acorn. The Regexp is simply not powerful enough as written.

**Comment by cheton** [comment]
There is a bug to the regular expression, it will pass the following code to the acorn JSX parser for your sample:

```
{variable}</0>" />
            <Trans>Hello, World!
```

**Comment by cheton** [comment]
Fixed in v2.6.2

**Comment by beheh** [comment]
Thanks for the fast fix, this is really great! I'm just having one last issue with JSX that contains a shorthand fragment (`<>like this</>`), it looks like acorn-jsx-walk can't handle that. I'll file a separate issue though.
